pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/SegmentGeneratorConfig.java

```diff
@@ -262,6 +266,21 @@ private void extractH3IndexConfigsFromTableConfig(TableConfig tableConfig) {
     }
   }
 
+  private void extractNoDictionaryColumnCompressionCodecConfigsFromTableConfig(TableConfig tableConfig) {
+    Map<String, String> rawIndexColumnsToCompressionTypeMap = new HashMap<>();
+    List<String> rawIndexColumns = new ArrayList<>();
+    List<FieldConfig> fieldConfigList = tableConfig.getFieldConfigList();
+    if (fieldConfigList != null) {
+      for (FieldConfig fieldConfig : fieldConfigList) {
+        if (fieldConfig.getEncodingType() == FieldConfig.EncodingType.RAW && fieldConfig.getNoDictionaryColumnCompressionCodec() != null) {
+          rawIndexColumns.add(fieldConfig.getName());
+          rawIndexColumnsToCompressionTypeMap.put(fieldConfig.getName(), fieldConfig.getNoDictionaryColumnCompressionCodec().name());
+        }
+      }
+    }
+    setRawIndexColumnCompressionType(rawIndexColumns, rawIndexColumnsToCompressionTypeMap);
```

Comment:
I don't think this still fixes the problem. At line 200, the following method will be called 

```
public void setRawIndexCompressionType(Map<String, ChunkCompressionType> rawIndexCompressionType) {
    _rawIndexCompressionType.clear();
    _rawIndexCompressionType.putAll(rawIndexCompressionType);
  }
```

When it is called again at line 281, clear() will be called and old compression config coming from noDictionaryConfig will be wiped out from it's first invocation. 

I think implementing the suggestion in https://github.com/apache/incubator-pinot/pull/6876#discussion_r626128003 is a simple fix. No need to call `setRawIndexColumnCompressionType`() from `extractNoDictionaryColumnCompressionCodecConfigsFromTableConfig`

Reply:
The current code won't be able to handle the compression config set through old way

Lets say column1 and column2 in an existing table T.

For column1, someone specified compression config through old way (noDictionaryConfig map)
For column2, someone specified compression config (SNAPPY, ZSTD) through new way (FieldConfig)

The current code won't be able to preserve the config of column1 which we need to handle until everything is migrated from existing way to FieldConfig

I haven't seen the old way of config being used so far at Li in production. But we don't know if someone in open source is using it or not. If they are, it will break things for them

Reply:
updated!